### PR TITLE
Address review feedback for wrapping and footnotes

### PR DESCRIPTION
**Pull request by leynos**
Centralize mutable state in the list, HTML-table, footnote, and paragraph-wrapping flows so the behaviour is easier to reason about from one place.

Fix the cargo-binstall Linux override, remove production `expect` usage from the reviewed paths, normalize whitespace-only wrapped lines, and add regression tests for mixed code-emphasis affixes, Unicode-width indentation, and indented multi-line HTML tables.

## Summary by Sourcery

Centralize mutable state across paragraph wrapping, footnote renumbering, list renumbering, and HTML table conversion to simplify control flow and improve robustness.

Bug Fixes:
- Fix cargo-binstall Linux override configuration to install binaries from the current directory.
- Normalize whitespace-only wrapped lines to emit consistent empty lines.
- Use Unicode display width when wrapping indented lines so full-width characters are handled correctly.
- Ensure HTML table conversion correctly handles indented multi-line HTML tables.
- Fix mixed code/emphasis affix handling so pending prefixes are cleared correctly in edge cases.

Enhancements:
- Refactor paragraph wrapping to encapsulate buffer and indentation handling inside ParagraphWriter.
- Refactor footnote renumbering to consolidate scan and numeric candidate state into a single DefinitionScanState struct.
- Refactor list renumbering state management with helper methods for resetting and computing next list numbers.
- Refactor HTML table parsing to manage buffer and depth via HtmlTableState and reuse it across conversion paths.
- Introduce a SplitContext::new constructor to simplify split context creation in wrapping code and tests.

Build:
- Adjust cargo-binstall Linux override to point bin-dir at the current directory instead of the binary path.

Tests:
- Add regression tests for whitespace-only line normalization in wrapping.
- Add regression tests for Unicode-width indentation handling in wrapping.
- Add regression tests for mixed code/emphasis affix handling.
- Add regression tests for indented multi-line HTML table conversion.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

~~~
Please address the comments from this code review:

## Overall Comments
- In `HtmlTableState::push_html_line` you increment `depth` using `TABLE_START_RE` on `trimmed` but decrement using `TABLE_END_RE` on the original `line`; consider applying both start and end regexes consistently on the same representation (trimmed or original) to avoid subtle mismatches with indented or oddly spaced tags.
- Now that `PrefixLine.prefix` is always a `String`, `prefix_line` unconditionally allocates even for simple matches; if performance on large files is a concern, you might consider retaining a borrowing representation for the common bullet/blockquote cases and only allocating when the prefix needs to be synthesized (e.g., footnotes).

## Individual Comments

### Comment 1
<location path="src/footnotes/renumber.rs" line_range="393-394" />
<code_context>
-        let new_number = assign_new_number(ctx.mapping, candidate.number, ctx.next_number);
-        let rewritten_rest = rewrite_tokens(&candidate.rest, ctx.mapping);
+fn finalize_numeric_candidates(state: &mut DefinitionScanState<'_>) {
+    for candidate in state.numeric_candidates.drain(..).rev() {
+        let new_number = assign_new_number(state.mapping, candidate.number, state.next_number);
+        let rewritten_rest = rewrite_tokens(&candidate.rest, state.mapping);
</code_context>
<issue_to_address>
**suggestion:** Consider draining numeric candidates in-place without reversing if ordering is not semantically required.

This now drains `numeric_candidates` in reverse, preserving the previous `into_iter().rev()` behavior. If that ordering isn’t required for correctness, consider iterating in insertion order and dropping the `.rev()`. If it *is* required (e.g., to avoid index churn while mutating), please add a short comment explaining the dependency on reverse order where this iteration is defined.

```suggestion
fn finalize_numeric_candidates(state: &mut DefinitionScanState<'_>) {
    for candidate in state.numeric_candidates.drain(..) {
```
</issue_to_address>

### Comment 2
<location path="src/wrap/tests.rs" line_range="303-306" />
<code_context>
 }

+#[test]
+fn wrap_text_normalizes_whitespace_only_lines() {
+    let input = vec![String::new(), "   ".to_string(), "\t\t".to_string()];
+    let wrapped = wrap_text(&input, 80);
+    assert_eq!(wrapped, vec![String::new(), String::new(), String::new()]);
+}
+
</code_context>
<issue_to_address>
**suggestion (testing):** Add a case with whitespace-only lines inside a paragraph to prove paragraph-splitting behavior.

Since `wrap_text` now treats `trim().is_empty()` as a paragraph break, it’d be helpful to cover a mixed case like `["foo", "   ", "bar"]`, asserting that the middle line normalizes to an empty string and that `foo` and `bar` end up in separate wrapped paragraphs. That directly exercises the paragraph-boundary behavior of `push_blank_line` and guards against whitespace normalization merging or dropping paragraphs.

Suggested implementation:

```rust
 }

 #[test]
 fn wrap_text_normalizes_whitespace_only_lines() {
     let input = vec![String::new(), "   ".to_string(), "\t\t".to_string()];
     let wrapped = wrap_text(&input, 80);
     assert_eq!(wrapped, vec![String::new(), String::new(), String::new()]);
 }

 #[test]
 fn wrap_text_treats_whitespace_only_lines_as_paragraph_breaks() {
     let input = vec![
         "foo".to_string(),
         "   ".to_string(),
         "bar".to_string(),
     ];
     let wrapped = wrap_text(&input, 80);

     // The whitespace-only line should normalize to an empty string and act as
     // a paragraph separator, so "foo" and "bar" remain in separate paragraphs.
     assert_eq!(
         wrapped,
         vec![
             "foo".to_string(),
             String::new(),
             "bar".to_string(),
         ]
     );
 }


```

If `wrap_text` is not already imported into this test module, ensure there is a `use` bringing it into scope (for example, `use crate::wrap::wrap_text;` or similar based on the existing conventions in `src/wrap/tests.rs`).
</issue_to_address>

### Comment 3
<location path="tests/table/convert_html.rs" line_range="82-78" />
<code_context>
 }
+
+#[test]
+fn converts_indented_multiline_html_table() {
+    let input = lines_vec![
+        "  <table>",
+        "    <tr><th>A</th><th>B</th></tr>",
+        "    <tr><td>1</td><td>2</td></tr>",
+        "  </table>",
+    ];
+    let expected = lines_vec![
+        "  | A | B |",
+        "  | --- | --- |",
+        "  | 1 | 2 |",
+    ];
+    assert_eq!(convert_html_tables(&input), expected);
+}
</code_context>
<issue_to_address>
**suggestion (testing):** Consider adding a test where an indented HTML table is followed by non-table HTML/content at the same indentation.

This test covers the core regression for indented multi-line tables and the refactored `HtmlTableState`. To better exercise the `buf`/`depth` state and `flush_raw`, please also add a case with:

- indented non-table content
- an indented `<table>…</table>` block
- more indented non-table content at the same indentation

and assert that only the table is converted and the surrounding lines pass through unchanged, verifying the `in_html()`/`flush_raw` transitions into and out of indented tables.
</issue_to_address>

### Comment 4
<location path="src/footnotes/renumber.rs" line_range="328" />
<code_context>
     let mut line = String::with_capacity(parts.prefix.len() + rewritten_rest.len() + 8);
     line.push_str(parts.prefix);
-    write!(&mut line, "[^{new_number}]:").expect("write to string cannot fail");
+    let _ = write!(&mut line, "[^{new_number}]:");
     line.push_str(&rewritten_rest);
     DefinitionLine {
</code_context>
<issue_to_address>
**issue (review_instructions):** Add tests demonstrating the bug this change addresses (ignoring write! errors instead of panicking) and verifying that renumbering behaves correctly when formatting succeeds.

Previously this code used expect("write to string cannot fail"), which would panic if formatting failed; now it ignores the Result from write!. This is a behavioural change intended to avoid panics and is effectively a bug fix. Add targeted tests that:
- Exercise renumbering of footnotes and numeric candidates to confirm no panics occur and the output lines are correctly formatted.
- If applicable, cover edge cases that could previously trigger the panic, to show the bug is fixed and behaviour remains correct.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*`

**Instructions:**
Bug fixes must be demonstrated by a test.

</details>
</issue_to_address>

### Comment 5
<location path="src/footnotes/renumber.rs" line_range="342" />
<code_context>
-    let rest_match = caps
-        .name("rest")
-        .expect("numeric list capture missing rest");
+    let num_match = caps.name("num")?;
+    let rest_match = caps.name("rest")?;
     let whitespace = line[num_match.end() + 1..rest_match.start()].to_string();
</code_context>
<issue_to_address>
**issue (review_instructions):** Add a regression test showing that malformed numeric list lines no longer panic due to missing regex captures.

numeric_candidate_from_line previously called expect on the "num" and "rest" captures, which would panic on malformed or unexpected input; now it returns None when captures are missing. This is a bug fix in error handling. Add a test that feeds a line matching NUMERIC_LIST_RE partially or unexpectedly (e.g., missing rest or number group) and assert that renumbering completes without panic and with the expected behaviour (either ignoring the line or handling it in a defined way).

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*`

**Instructions:**
Bug fixes must be demonstrated by a test.

</details>
</issue_to_address>
~~~

File: src/wrap/tests.rs [hunk 1]
```diff
@@ -308,6 +299,20 @@ fn wrap_text_preserves_code_spans() {
...
+    let wrapped = wrap_text(&input, 80);
+    assert_eq!(wrapped, vec![String::new(), String::new(), String::new()]);
+}
+
+#[test]
+fn wrap_text_uses_display_width_for_unicode_indent() {
+    let input = vec!["　a".to_string()];
+    let wrapped = wrap_text(&input, 2);
+    assert_eq!(wrapped, vec!["　a".to_string()]);
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/wrap/tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

[the remark this answers is not in the document]
Comment (leynos) [reply src/wrap/tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/footnotes/renumber/definitions.rs [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+) -> bool {
+    if skip_numeric_conversion {
+        return false;
+    }
+    numeric_range.is_some_and(|(start, end)| index >= start && index < end)
+}
+
+fn definition_line_from_parts(
+    index: usize,
+    parts: super::super::parsing::DefinitionParts<'_>,
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/footnotes/renumber/definitions.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai pause

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (4 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing | ⚠️ Warning | The PR introduces substantial architectural refactorings without adequate direct unit test coverage for critical refactored components. | Add focused unit tests for definitions module functions, ParagraphWriter methods, and prefix\_line() edge cases, including snapshot tests for text transformations. |
| User-Facing Documentation | ⚠️ Warning | The pull request introduces user-facing behavioural improvements that lack documentation in CHANGELOG.md or README.md, including whitespace normalisation, Unicode display width calculations, HTML table handling, and cargo-binstall configuration. | Update CHANGELOG.md under the 'Fixed' section with entries documenting whitespace normalisation, Unicode display width calculations, indented HTML table handling, and cargo-binstall directory configuration. |
| Developer Documentation | ⚠️ Warning | Pull request introduces five significant internal APIs and architectural abstractions without corresponding documentation in docs/architecture.md. | Update docs/architecture.md to document ParagraphWriter, HtmlTableState, DefinitionScanState, ListState, and definitions submodule with their responsibilities and pipeline integration. |
| Module-Level Documentation | ⚠️ Warning | The file tests/wrap\_unit.rs lacks a module-level docstring, starting directly with use statements instead of documentation explaining the module's purpose. | Add a module-level docstring to tests/wrap\_unit.rs before the use statements, explaining that it contains unit tests for text wrapping functionality. |

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 errors, 5 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Footnote tests not parameterised, property tests missing, HTML depth test incomplete, ListState methods untested. | Parameterise footnote tests with rstest; add proptest for renumbering invariants; test nested table depth tracking; add unit tests for ListState methods. |
| Module-Level Documentation | ❌ Error | Module `definitions` declared in `src/footnotes/renumber.rs` (line 7) lacks a module-level docstring explaining its purpose and relationship to other components. | Add a documentation comment above `mod definitions;` explaining that it owns definition parsing/reordering machinery, delegated from the parent module to keep files readable. |
| Unit Architecture | ❌ Error | DefinitionUpdates::definitions doc claims "in scan order" but finalize\_numeric\_candidates drains in reverse order, violating stated architectural contract. | Sort DefinitionUpdates::definitions by DefinitionLine::index before returning from collect\_definition\_updates, or revise Rustdoc to document actual reverse-drain ordering. |
| User-Facing Documentation | ⚠️ Warning | Three user-facing fixes lack CHANGELOG.md entries: Unicode display-width wrapping, indented multi-line HTML table conversion, and cargo-binstall Linux bin-dir fix. | Add three Fixed bullets to CHANGELOG.md documenting Unicode display-width indentation handling, indented HTML table conversion, and cargo-binstall Linux binary installation directory. |
| Developer Documentation | ⚠️ Warning | CHANGELOG.md lacks three required user-facing fixes: Unicode display-width wrapping, indented multi-line HTML table conversion, and cargo-binstall Linux binary-directory fix. | Add three Fixed bullets to CHANGELOG.md documenting Unicode display-width wrapping, indented multi-line HTML table conversion, and cargo-binstall Linux fix. |
| Testing (Unit And Behavioural) | ⚠️ Warning | Property-based testing for renumbering invariants was requested but not implemented. Two renumber\_footnotes tests remain separate instead of parameterised with #\[rstest\]. | Add proptest for renumbering invariants. Parameterise renumber\_footnotes tests into single #\[rstest\] function with two #\[case\] attributes. |
| Testing (Property / Proof) | ⚠️ Warning | PR introduces stateful renumbering with invariants across mixed orderings but lacks proptest; review explicitly requested property tests for dense sequential numbering—unaddressed. | Add proptest generating mixed footnote references, repeated references, numeric candidates, and fenced blocks; assert dense sequential renumbering and untouched fenced content. |
| Domain Architecture | ⚠️ Warning | Domain command collect\_definition\_updates claims "in scan order" but finalize\_numeric\_candidates reverses candidate order, violating documented invariant contract. | Sort DefinitionUpdates::definitions by index before return, or rewrite Rustdoc to match actual reverse-order numeric candidate appending behaviour. |

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 errors, 5 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Footnote tests not parameterised, property tests missing, HTML depth test incomplete, ListState methods untested. | Parameterise footnote tests with rstest; add proptest for renumbering invariants; test nested table depth tracking; add unit tests for ListState methods. |
| Module-Level Documentation | ❌ Error | Module `definitions` declared in `src/footnotes/renumber.rs` (line 7) lacks a module-level docstring explaining its purpose and relationship to other components. | Add a documentation comment above `mod definitions;` explaining that it owns definition parsing/reordering machinery, delegated from the parent module to keep files readable. |
| Unit Architecture | ❌ Error | DefinitionUpdates::definitions doc claims "in scan order" but finalize\_numeric\_candidates drains in reverse order, violating stated architectural contract. | Sort DefinitionUpdates::definitions by DefinitionLine::index before returning from collect\_definition\_updates, or revise Rustdoc to document actual reverse-drain ordering. |
| User-Facing Documentation | ⚠️ Warning | Three user-facing fixes lack CHANGELOG.md entries: Unicode display-width wrapping, indented multi-line HTML table conversion, and cargo-binstall Linux bin-dir fix. | Add three Fixed bullets to CHANGELOG.md documenting Unicode display-width indentation handling, indented HTML table conversion, and cargo-binstall Linux binary installation directory. |
| Developer Documentation | ⚠️ Warning | CHANGELOG.md lacks three required user-facing fixes: Unicode display-width wrapping, indented multi-line HTML table conversion, and cargo-binstall Linux binary-directory fix. | Add three Fixed bullets to CHANGELOG.md documenting Unicode display-width wrapping, indented multi-line HTML table conversion, and cargo-binstall Linux fix. |
| Testing (Unit And Behavioural) | ⚠️ Warning | Property-based testing for renumbering invariants was requested but not implemented. Two renumber\_footnotes tests remain separate instead of parameterised with #\[rstest\]. | Add proptest for renumbering invariants. Parameterise renumber\_footnotes tests into single #\[rstest\] function with two #\[case\] attributes. |
| Testing (Property / Proof) | ⚠️ Warning | PR introduces stateful renumbering with invariants across mixed orderings but lacks proptest; review explicitly requested property tests for dense sequential numbering—unaddressed. | Add proptest generating mixed footnote references, repeated references, numeric candidates, and fenced blocks; assert dense sequential renumbering and untouched fenced content. |
| Domain Architecture | ⚠️ Warning | Domain command collect\_definition\_updates claims "in scan order" but finalize\_numeric\_candidates reverses candidate order, violating documented invariant contract. | Sort DefinitionUpdates::definitions by index before return, or rewrite Rustdoc to match actual reverse-order numeric candidate appending behaviour. |

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 errors, 7 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Tests are substantive but fail to cover three unaddressed review comments: missing whitespace normalisation in handle\_pending\_continuation; no tracing::warn for reorder skips; unsafe slice indexing. | Apply whitespace normalisation to handle\_pending\_continuation; add tracing::warn for reorder skips; add bounds checking in numeric\_candidate\_from\_line. |
| Unit Architecture | ❌ Error | Four documented review items unresolved: docs line-wrapping, whitespace normalisation, diagnostic warning, and bounds check. | Apply outstanding review fixes: reflow docs/architecture.md to 80 cols, add whitespace normalisation, add tracing::warn, guard slice extraction with saturating\_add. |
| Security And Privacy | ❌ Error | Three review issues unresolved: missing bounds check for slice indexing at definitions.rs:161, missing diagnostic warning for reorder skip, missing whitespace normalisation in pending prefix path. | Implement saturating\_add bounds check, add tracing::warn for reorder skip, apply whitespace-only normalisation before push\_verbatim calls. |
| User-Facing Documentation | ⚠️ Warning | Four user-facing features lack users-guide.md documentation: --code-emphasis flag, whitespace-only line normalisation, Unicode display width indentation, and indented multi-line HTML table conversion. | Document the four user-facing features in docs/users-guide.md: add --code-emphasis section, update wrapping section for whitespace-only lines and Unicode display width, and add HTML table conversion details. |
| Developer Documentation | ⚠️ Warning | New internal APIs (HtmlTableState, DefinitionScanState, ListState) not documented in developers-guide.md. architecture.md lines 596–597 exceed 80-column limit. Three review suggestions unaddressed. | Document new APIs in developers-guide.md; wrap architecture.md lines 596–597; add tracing::warn!, bounds check, and whitespace normalisation per review comments. |
| Testing (Unit And Behavioural) | ⚠️ Warning | Three review feedback items remain unimplemented: whitespace normalisation in pending\_prefix, diagnostic for reorder mismatch, and bounds check for regex captures. | Implement pending\_prefix whitespace normalisation, add tracing::warn! for reorder mismatch, add bounds validation, and reflow architecture.md to 80 columns. |
| Testing (Property / Proof) | ⚠️ Warning | PR introduces invariants for ListState indent-stack pruning and HtmlTableState depth tracking but only adds proptest for footnote renumbering, not these state machine invariants. | Add proptest property tests for ListState::next\_number with varying indent sequences and HtmlTableState::push\_html\_line with nested table structures. |
| Domain Architecture | ⚠️ Warning | Four unresolved review items: whitespace normalisation in wrap.rs, missing diagnostic in definitions.rs, missing bounds check, and documentation exceeding 80-column limit. | Fix whitespace normalisation, add tracing::warn for reorder mismatch, add saturating bounds check, and reflow docs to ≤80 columns. |
| Observability | ⚠️ Warning | Missing observability: reorder\_definition\_block lacks tracing::warn on length mismatch; handle\_pending\_continuation lacks whitespace normalisation; numeric\_candidate\_from\_line lacks bounds check. | Add tracing::warn diagnostic when reorder skipped; apply whitespace normalisation in pending-prefix; add saturating\_add bounds check. |
| Architectural Complexity And Maintainability | ⚠️ Warning | Four review comments unresolved: pending-prefix whitespace normalisation, reorder\_definition\_block diagnostic, numeric\_candidate\_from\_line bounds check, and docs/architecture.md column wrapping. | Implement pending review feedback: whitespace handling in handle\_pending\_continuation; tracing::warn in reorder\_definition\_block; saturating\_add in numeric\_candidate\_from\_line; reflow docs/architecture.md to 80 columns. |

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### src/footnotes/renumber/definitions.rs (1)

`294-382`: _🛠️ Refactor suggestion_ | _🟠 Major_ | _🏗️ Heavy lift_
**Decompose `reorder_definition_block` to cut cognitive complexity and enforce CQRS boundaries.**
Split segmentation, ordering, blank-line migration, and final writeback into dedicated helpers. Keep one coordinator function that orchestrates those steps and performs the single mutation point.
As per coding guidelines, "Seek to keep the cognitive complexity of functions no more than 9" and "Adhere to single responsibility and CQRS."

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@src/footnotes/renumber/definitions.rs` around lines 294 - 382, The function
reorder_definition_block is too large and mixes segmentation, ordering,
blank-line migration, and the final writeback; decompose it into small helpers
to reduce cognitive complexity and enforce CQRS: add pure helpers like
collect_header_positions(lines, start, end) (uses parse_definition),
build_def_lookup(definitions, start, end) (returns
HashMap<usize,&DefinitionLine>), build_segments(lines, header_positions,
def_lookup, end) (uses definition_segment_end and is_definition_continuation to
produce Vec<(new_number, index, Vec<String>)>), migrate_first_leading(segments:
&mut Vec<...>) (extracts leading blanks from first segment into a Vec<String>),
and a single coordinator reorder_definition_block that calls these helpers,
sorts segments, composes reordered, and performs the single mutation at
lines[start..end].clone_from_slice(&reordered); keep existing debug_assert/warn
behavior but move logging to coordinator. Ensure each helper is small, pure
where possible, and reference DefinitionLine for types.

### ♻️ Duplicate comments (1)

#### tests/wrap_unit/code_spans.rs (1)

`88-93`: _⚠️ Potential issue_ | _🟡 Minor_ | _⚡ Quick win_
**Restore the lossless-content assertion in this regression guard.**
Add a reconstruction check so the test fails when wrapping drops or rewrites tokens while still satisfying width and first-break expectations.

##### 🐛 Proposed fix

```diff
assert!(
         wrapped[0].ends_with("`.toml`,") || wrapped[0].ends_with("`.json`,"),
         "expected first line to break inside the code-span list, got: {:?}",
         wrapped[0]
     );
+    assert_eq!(wrapped.join(" "), input[0]);
```

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@tests/wrap_unit/code_spans.rs` around lines 88 - 93, Add a lossless-content
assertion after the existing first-line check: reconstruct the wrapped output by
joining the wrapped lines with '\n' (or the original line break used in the
test) and assert that this reconstructed string equals the original source
string used to produce `wrapped` (the input string/variable used earlier in this
test, e.g. `s` or `input`). This ensures the wrapping did not drop or rewrite
tokens while still meeting width and break expectations; place the assertion
immediately after the existing `assert!( wrapped[0].... )` block referencing
`wrapped` and the original input variable.

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### src/wrap/inline/postprocess.rs (1)

`169-404`: _🛠️ Refactor suggestion_ | _🟠 Major_ | _⚡ Quick win_
**Decompose this module to satisfy the 400-line limit.**
Split the in-file test module into a sibling test file or extract helper/test sections so this file stays at or under 400 lines. Line-count is now over the repository cap.
As per coding guidelines, "`Files must not exceed 400 lines in length`".

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@src/wrap/inline/postprocess.rs` around lines 169 - 404, The test module in
postprocess.rs exceeds the 400-line limit; extract the entire #[cfg(test)] mod
tests block into a new sibling test file (e.g., postprocess_tests.rs) and remove
it from postprocess.rs. In the new file keep the same imports and test bodies
but adjust the path imports to reference the parent module (use
super::{super::fragment::{FragmentKind, InlineFragment}, *}; or appropriate
crate:: path) so tests still access InlineFragment, FragmentKind,
merge_whitespace_only_lines, rebalance_atomic_tails, line_width, etc.; finally
run cargo test to verify and delete the original tests block to bring
postprocess.rs under 400 lines.

#### src/html.rs (1)

`351-419`: _🛠️ Refactor suggestion_ | _🟠 Major_ | _⚡ Quick win_
**Split this module to comply with the 400-line cap.**
Extract the test module (or part of the conversion helpers) into separate files and keep this file at or under 400 lines.
As per coding guidelines, "`Files must not exceed 400 lines in length`".

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@src/html.rs` around lines 351 - 419, This file exceeds the 400-line limit:
split out the test helpers into a new test module/file to keep src/html.rs under
400 lines. Move the #[cfg(test)] mod tests (or at least the large proptest_tests
submodule and/or helper functions like HtmlTableState tests) into a new file
(e.g., tests/html_tests.rs) and re-export or keep private test-only items;
ensure the tests still import is_element, is_table_cell and HtmlTableState from
the parent module (use super:: or crate::html as needed) so tests compile and
run unchanged. Update module declarations so proptest_tests, element_detection
and table_cell_detection reference the same symbols (is_element, is_table_cell,
HtmlTableState) and run cargo test to verify.

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### src/html.rs (2)

`289-313`: _🛠️ Refactor suggestion_ | _🟠 Major_ | _⚡ Quick win_
**Merge the two doc-comment blocks and move `#[must_use]` below them.**
`convert_html_tables` carries two separate `///` blocks with `#[must_use]` wedged between them (lines 289‑292, then 293, then 294‑312). Rustdoc concatenates both, producing duplicated, contradictory prose, and the attribute breaks the "attributes after doc comments" rule. Fold the descriptions into one block and place `#[must_use]` immediately above `pub fn`.
As per coding guidelines: "Place function attributes **after** doc comments."

##### ♻️ Proposed consolidation

```diff
-/// Processes Markdown lines and converts embedded HTML tables to Markdown.
-///
-/// Fenced code blocks are left untouched, allowing raw HTML examples to be
-/// documented without modification.
-#[must_use]
-/// Converts HTML tables embedded in Markdown lines to Markdown table syntax.
-///
-/// Scans the input lines, detects HTML table blocks outside of fenced code blocks, and replaces
-/// them with equivalent Markdown tables. Fenced code blocks are left unmodified. Handles nested
-/// tables and preserves original line formatting outside of tables.
+/// Converts HTML tables embedded in Markdown lines to Markdown table syntax.
+///
+/// Scans the input lines, detects HTML table blocks outside of fenced code blocks, and replaces
+/// them with equivalent Markdown tables. Fenced code blocks are left untouched, allowing raw HTML
+/// examples to be documented without modification. Handles nested tables and preserves original
+/// line formatting outside of tables.
 ///
 /// # Examples
 ///
 /// ```no_run
 /// use mdtablefix::convert_html_tables;
 /// let lines = vec![
 ///     "<table>".to_string(),
 ///     "  <tr><th>Header</th></tr>".to_string(),
 ///     "  <tr><td>Cell</td></tr>".to_string(),
 ///     "</table>".to_string(),
 /// ];
 /// let result = convert_html_tables(&lines);
 /// assert!(result[0].starts_with("| Header |"));
 /// ```
+#[must_use]
 pub fn convert_html_tables(lines: &[String]) -> Vec<String> {
```

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@src/html.rs` around lines 289 - 313, The file has two separated doc-comment
blocks for convert_html_tables with #[must_use] inserted between them; merge the
two /// documentation blocks into a single contiguous doc comment describing the
function and examples, then move the #[must_use] attribute so it sits
immediately above pub fn convert_html_tables(...) (i.e., doc comments first,
then #[must_use], then the function signature) to follow the guideline that
attributes come after doc comments.

`228-236`: _⚠️ Potential issue_ | _🟡 Minor_ | _⚡ Quick win_
**Make start/end counting symmetric (or document the one-tag-per-line contract).**
`TABLE_START_RE` is anchored (`^<table...`), so `find_iter(trimmed).count()` can only ever increment by 0/1 per line, while `TABLE_END_RE` is unanchored (`</table>`) and `find_iter(trimmed).count()` can decrement multiple times for `</table></table>` on one line—desynchronising `depth` for inputs like `<table><table>` followed by `</table></table>`. There are no lines in the current repository containing 2+ `<table` or 2+ `</table` occurrences, so existing fixtures won’t exercise this, but the trap remains for other inputs.
Replace the start counting with an explicit boolean, and either count all unanchored start tags too (remove the `^`) or document the single-start-tag-per-line + single-end-tag-per-line assumption.

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@src/html.rs` around lines 228 - 236, push_html_line currently increments
depth using TABLE_START_RE which is anchored so it can only ever add 0/1 while
it decrements by counting all TABLE_END_RE matches; make counting symmetric by
changing the start-counting to count all start-tag occurrences instead of an
anchored boolean (i.e., use TABLE_START_RE.find_iter(trimmed).count() after
removing the leading-anchor from the regex) or alternatively treat both sides as
booleans (use is_match for both) and document the one-tag-per-line contract;
update the logic in push_html_line (function name) referencing TABLE_START_RE
and TABLE_END_RE so depth is adjusted with matching semantics for starts and
ends.

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai review
